opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -208,8 +209,38 @@ def set_attribute(self, key: str, value: types.AttributeValue) -> None:
         if has_ended:
             logger.warning("Setting attribute on ended span.")
             return
+
+        if not isinstance(value, (int, float, bool, str, list, tuple)):
+            logger.warning("invalid type for attribute value")
+            return
+        if isinstance(value, (list, tuple)) and len(value) > 0:
+            return_code = self._check_sequence(value)
+            if return_code:
+                logger.warning("%s in attribute value sequence", return_code)
+                return
+
         self.attributes[key] = value
 
+    @staticmethod
+    def _check_sequence(sequence: (list, tuple)) -> Union[str, None]:
```

Comment:
I created a method in my suggestion as an example only. Nevertheless, making it private as it is here is the right approach. I personally never create a method when it is called only once (I just put the contents of the method where it would be called). In this case it is justified to create a method since you are calling it in this module and also in the test cases. I prefer also this approach of having it in the class and not in a separate util module since this method is only used here, the narrower the scope where the method is available the better.

Reply:
It looks like I took some of your feedback a bit too literally...  I'm still figuring out Python best practices as well as this project's best practices so I wasn't sure if your example was a suggestion to separate out the logic. I'll keep that in mind for future feedback.

I *did* think it belonged in a separate method since the validation logic for one case was longer than the rest of the method. I just originally left it in the method because I wasn't sure where the logic belonged.  This makes the most sense to me but I'm happy to take any feedback on it.